tools/shared.py

```diff
@@ -185,7 +187,13 @@ def run_process(cmd, check=True, input=None, *args, **kw):
   # Python 2 compatibility: Introduce Python 3 subprocess.run-like behavior
   if input is not None:
     kw['stdin'] = subprocess.PIPE
-  proc = Popen(cmd, *args, **kw)
+  try:
+    proc = Popen(cmd, *args, **kw)
+  except OSError as e:
+    if MSYS and e.errno is errno.ENOEXEC:
+      return run_process([MSYS, '-i'] + cmd, check, input, *args, **kw)
+    else:
+      raise e
```

Comment:
What is MSYS here?  Some kind of executable that look at the #! line?

I have to say this change looks pretty hacky.  I hope we can find a cleaner way to do this.

Reply:
Is this running msys.bat?  i.e. https://github.com/msysgit/msysgit/blob/master/msys.bat?   Can the value of MSYSTEM env var take other value that might not correspond to executables?

Why the `-i`?  If this is bash then we probably don't want an interactive shell do we?

I'd almost rather just unconditionally inject "bash.exe" as the beginning of the command line if `MSYSTEM` is set.

Reply:
Nowadays i am using the Bash that comes with [git-for-windows](https://github.com/git-for-windows/git/releases).

MSYS here is the value of `$SHELL` when `$MSYSTEM` is defined, otherwise `False`

`$SHELL` does usually points to `/usr/bin/bash`

> Why the -i? If this is bash then we probably don't want an interactive shell do we?

I had issues by not using `-i`, however i have been patching emconfigure.py on my end since ages ago and don't recall atm what exact issue was that.

> I'd almost rather just unconditionally inject "bash.exe" ...

Actually this is something i had thought about doing, but i had considered it could turn dangerous and/or overkill, for example there is no need to spawn a new bash to execute e.g. llvm-nm.exe or other such Windows executables.

Also, I think doing so will involve that I/we track down every `if WINDOWS:` usages and related logic around to ensure spawning through bash won't have side-effects/malfunctions, and likely this is not worth the trouble, so i took the pragmatic approach of only attempting to spawn through bash when we do get an actual `ENOEXEC`